### DISCUSSION TITLE
Cannot use simple models derived from str (with Pydantic RootModel) as query params

### DISCUSSION THREAD

**Discussion by WarpedPixel** [fastapi/fastapi#11101] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# fail.py
# uvicorn fail:app --reload

from fastapi import FastAPI
from pydantic import RootModel

SimpleID = RootModel[str]

app = FastAPI()

@app.get("/test1")
async def t1(param1: SimpleID):     # BUG: expected to be query param, but it is body
    return {"message": f"Hello {param1}"}

# @app.get("/test2")
# async def t2(param1: Annotated[SimpleID, Query()]): # this workaround does NOT work
#     return {"message": f"Hello {param1}"}

@app.get("/test3")
async def t3(param1: SimpleID, param2: str):  # param2 is query param as expected, param1 is NOT
    return {"message": f"Hello {param1} {param2}"}
```

### Description

Simple types should be usable as query params. Pydantic provides `RootModel` (instead of `BaseModel`) precisely to implement types with rich validation based on native types like `list` or `str` (instead of the more typical object/dict). Types built with `RootModel` do not seem to work as query params, *even if it is functionally equivalent* to a simple type like `str`.
Furthermore, we cannot force them to work as query params with `Query()` annotations either.
**NOTE:** the example is overly simplified, my RootModel-derived types do much more in terms of validation.

### Operating System

macOS

### FastAPI Version

0.108.0

### Pydantic Version

2.5.3

### Python Version

Python 3.11.4

### Additional Context

I am not sure this is possible, but I would like for FastAPI to understand RootModel derived types, if the root type is one of the types that can be used as query param, then it should work as such.
The workaround in my case is to use `str` parameters in my FastAPI calls then manually convert to the rich Pydantic type that does validation. Which kind of defeats the purpose of FastAPI rich type annotations.

**Comment by Kludex** [comment, marked as answer]
https://github.com/tiangolo/fastapi/issues/11134
